lib/galaxy/tools/parameters/validation.py

```diff
@@ -103,20 +121,54 @@ class InRangeValidator( Validator ):
         ...
     ValueError: Not gonna happen
     """
+
     @classmethod
     def from_element( cls, param, elem ):
-        return cls( elem.get( 'message', None ), elem.get( 'min' ), elem.get( 'max' ) )
-    def __init__( self, message, range_min, range_max ):
+        return cls( elem.get( 'message', None ), elem.get( 'min' ),
+                    elem.get( 'max' ), elem.get( 'strict_min', 'false' ), 
+                    elem.get( 'strict_max', 'false' ) )
+
+    def __init__( self, message, range_min, range_max, strict_min=False, strict_max=False ):
+        """
+        When the optional strict_min and strict_max attributes are set
+        to true, the range excludes the end points (i.e., min < value < max),
+        while if set to False ( the default), then range includes the end points
+        (1.e., min <= value <= max).  Combinations of strict_min and strict_max
+        values are allowed.
+        """
         self.min = float( range_min if range_min is not None else '-inf' )
+        self.strict_min = util.asbool( strict_min )
         self.max = float( range_max if range_max is not None else 'inf' )
+        self.strict_max = util.asbool( strict_max )
         assert self.min <= self.max, 'min must be less than or equal to max'
         # Remove unneeded 0s and decimal from floats to make message pretty.
         self_min_str = str( self.min ).rstrip( '0' ).rstrip( '.' )
         self_max_str = str( self.max ).rstrip( '0' ).rstrip( '.' )
-        self.message = message or "Value must be between %s and %s" % ( self_min_str, self_max_str )
+        op1 = '>='
+        op2 = '<='
+        if self.strict_min and self.strict_max:
+            op1 = '>'
+            op2 = '<'
+        if self.strict_min:
+            op1 = '>'
+        elif self.strict_max:
+            op2 = '<'
+        self.message = message or "Value must be %s %s and %s %s" % ( op1, self_min_str, op2, self_max_str )
+
     def validate( self, value, history=None ):
-        if not( self.min <= float( value ) <= self.max ):
-            raise ValueError( self.message )
+        if self.strict_min and self.strict_max:
+            if not( self.min < float( value ) < self.max ):
+                raise ValueError( self.message )
+        elif self.strict_min and not self.strict_max:
+            if not( self.min < float( value ) <= self.max ):
+                raise ValueError( self.message )
+        elif not self.strict_min and self.strict_max:
+            if not( self.min <= float( value ) < self.max ):
+                raise ValueError( self.message )
+        elif not self.strict_min and not self.strict_max:
+            if not( self.min <= float( value ) <= self.max ):
+                raise ValueError( self.message )
```

Comment:
@nsoranzo thanks for the simplified validation logic.  I just committed it in another addition.